pandas/io/packers.py

```diff
@@ -506,11 +526,17 @@ def decode(obj):
         dtype = dtype_for(obj[u'dtype'])
         pd_dtype = pandas_dtype(dtype)
         np_dtype = pandas_dtype(dtype).base
+
+        ctor_dtype = np_dtype
+        if is_categorical_dtype(pd_dtype):
+            # Series ctor doesn't take dtype with categorical
+            ctor_dtype = None
```

Comment:
I need to start a new thread on this line of code.

So, if I understood, it shouldn't be necessary to special case the `dtype`? If I remove the `dtype=ctor_dtype` change (and leave it set to `dtype=np_dtype`), then it does not work. I get:

```
ValueError: cannot specify a dtype with a Categorical unless dtype='category'
```

I don't currently understand what is being asked for here.